### PR TITLE
update azure e2e tests

### PR DESCRIPTION
**Pull request by aramase**

### CODE REVIEW

**Comment by aramase** [comment]
/hold

to validate after https://github.com/kubernetes/test-infra/pull/17073 is merged to enable presubmit job for e2e-azure

**Comment by aramase** [comment]
/hold cancel 

`pull-secrets-store-csi-driver-e2e-azure` has triggered and run successfully.

@ritazh this is ready for review

File: test/bats/azure.bats [hunk 1]
```diff
@@ -6,13 +6,16 @@ BATS_TESTS_DIR=test/bats/tests
...
-export SECRET_NAME=secret1
-export KEY_NAME=key1
-export SECRET_NAME=secret1
-export SECRET_VERSION=""
+export KEYVAULT_NAME=${KEYVAULT_NAME:-csi-secrets-store-e2e}
+export SECRET_NAME=${KEYVAULT_SECRET_NAME:-secret1}
+export SECRET_VERSION=${KEYVAULT_SECRET_VERSION:-""}
+export SECRET_VALUE=${KEYVAULT_SECRET_VALUE:-"test"}
+export KEY_NAME=${KEYVAULT_KEY_NAME:-key1}
+export KEY_VERSION=${KEYVAULT_KEY_NAME:-7cc095105411491b84fe1b92ebbcf01a}
```
Comment (ritazh) [inline test/bats/azure.bats:17]: s/KEYVAULT_KEY_NAME/KEYVAULT_KEY_VERION

Comment (aramase) [reply test/bats/azure.bats]: thank you, updated!

File: test/bats/azure.bats [hunk 2]
```diff
@@ -6,13 +6,16 @@ BATS_TESTS_DIR=test/bats/tests
...
-export SECRET_NAME=secret1
-export KEY_NAME=key1
-export SECRET_NAME=secret1
-export SECRET_VERSION=""
+export KEYVAULT_NAME=${KEYVAULT_NAME:-csi-secrets-store-e2e}
+export SECRET_NAME=${KEYVAULT_SECRET_NAME:-secret1}
+export SECRET_VERSION=${KEYVAULT_SECRET_VERSION:-""}
+export SECRET_VALUE=${KEYVAULT_SECRET_VALUE:-"test"}
+export KEY_NAME=${KEYVAULT_KEY_NAME:-key1}
+export KEY_VERSION=${KEY_VERSION:-7cc095105411491b84fe1b92ebbcf01a}
```
Comment (ritazh) [inline test/bats/azure.bats:17]: nit: to be consistent, shouldnt this be `KEYVAULT_KEY_VERSION`

Comment (aramase) [reply test/bats/azure.bats]: updated!

**Pull request merged**